dll/win32/msafd/misc/dllmain.c

```diff
@@ -1389,6 +1389,9 @@ WSPSelect(IN int nfds,
     HeapFree( GlobalHeap, 0, PollBuffer );
     NtClose( SockEvent );
 
+    /* Allow APC to be processed */
+    SleepEx(0, TRUE);
```

Comment:
I wonder why this call is still needed, it feels a bit hackish

Reply:
It's due to the nature of APC in Windows. APC requires alertable wait state to be called.

> When a user-mode APC is queued, the thread to which it is queued is not directed to call the APC function unless it is in an alertable state.

[https://learn.microsoft.com/en-us/windows/win32/sync/asynchronous-procedure-calls](https://learn.microsoft.com/en-us/windows/win32/sync/asynchronous-procedure-calls)